Implement Travis CI integration with PHPCS/PHPCBF and reformat of legacy code to pass tests.

johangant opened the pull request:
This PR aims to:

1. Introduce Drupal coder module to the project (for PHPCS standards definitions) 
2. Introduce a new Bash script to the project root that can be used locally or in a Docker container to run static analysis of custom code. Its scope is limited to custom module + theme PHP code, so no Twig checks in there.
3. Reformat of failing legacy code so that it passes tests. No functional changes are expected here. The vast majority are annotation tweaks and some of these can potentially be relaxed later through the defined smells to ignore in `phpcs.sh`.
4. Amended Travis CI config to stop caching the entirety of `~/.composer` between builds as it causes problems with new projects added to `composer.json` in the docker container's installation phase. `~/.composer/cache` achieves similar results without affecting what is defined in `composer.lock` which is what `composer install` uses when invoked.

Functional tests performed are:

- Site bootstrap check before/after changes + `drush cr`
- Clicked into a few top level landing pages as logged in admin + anonymous user for any clear breaks, none found.

kirkkala commented on drupal/web/modules/custom/fsa_es/src/Plugin/Block/ViewsExposedFilterSearchBlock.php:
```diff
@@ -131,7 +140,13 @@ function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+      'post_route_name',
+      'location_context',
+      'disable_ajax',
+      'hide_submit_button'
```
Missing the comma after last value?

johangant replied:
Agreed, good spot!

johangant commented:
@kirkkala all Normalizer classes now use `{@inheritdoc}` for consistency with base Symfony class signatures, so far less stress/maintenance for our custom code.

Spotted and fixed a bug I had introduced as well where `vid` should take a string for Taxonomy vocabulary machine name rather than an integer for a numerical id value. I blame inconsistent/confusing Drupal 7 > Drupal 8 refactoring :)

I think this should be ready to merge whenever you're happy to do so 👍

johangant commented:
Oh, missed your questions further up. I'll review/address now

johangant commented on drupal/web/modules/custom/fsa_es/fsa_es.module:
```diff
@@ -122,8 +129,10 @@ function fsa_es_preprocess_views_view(&$variables) {
+        /*
+         * If there are no results in "search_global_all", add directly to
+         * "rows" element
+         */
```
No worries. If this is more consistent/helpful I'll change this.

kirkkala commented:
Nice @johangant starting to look good and Travis passes :)

However there was still few missing commas from the last array items and unused use statements. I branched a commit to fix those: https://github.com/wunderio/client-UK-FSA-beta/commit/9e381146138b3f78cc5d8951bda1323eb1cb58d2
